### ISSUE TITLE
Only one graph at a time reads tokens correctly

### ISSUE THREAD

**Issue by jdanish** [netcreateorg/nc-multiplex#3]
This became an issue with the pivot to netcreate-itest, and remains an issue in #2 

A typical usage pattern is to open multiple graphs at once. Let's call them graph1 and graph2.
Previously, you could login to both using appropriately generated tokens, and edit them simultaneously from different browsers / machines.

As a reminder, before running nc-multiplex, one runs package or dev on netcreate-itest. Let's assume that was done with graph1.

Now, if you use multiplex, the graph1 tokens work, but not graph2.

If you kill everything and go to netcreate-itest and run package with graph 2, then restart multiplex, graph1 no-longer accepts tokens but graph2 does. If I run graph1 and graph2 separately, one at a time in netcreate-itest they both work fine. So, it appears the issue is caused somewhere in the multiplex launching / communicating, not in netreate-itest (I know I shouldn't guess, but ....).

Also note that there are no errors in the console - just the normal on-screen error for an incorrect token, though the token should be correct.

**Comment by daveseah** [comment]
Hey Joshua,

could you provide some screenshots showing me the command you use to launch nc-multiplex and the specific errors on screen? That will help me figure out exactly what part of the code to look like (I'm not familiar with how multiplex is deployed in the field, so I have been looking at the wrong place)

**Comment by jdanish** [comment]
This is the image of starting it up:

<img width="904" alt="Screenshot 2023-07-25 at 9 11 57 PM" src="https://github.com/netcreateorg/nc-multiplex/assets/5529898/4d8fb6d3-292c-47d8-b020-583d16553547">

Then I go to http://localhost/manage and start 2 graphs:

<img width="1706" alt="Screenshot 2023-07-25 at 9 13 24 PM" src="https://github.com/netcreateorg/nc-multiplex/assets/5529898/067e6078-d1e0-48ef-b2d2-3300635642fc">

I try and generate tokens, but no go:

<img width="1695" alt="Screenshot 2023-07-25 at 9 14 03 PM" src="https://github.com/netcreateorg/nc-multiplex/assets/5529898/48e6a96a-2b20-45aa-a8f7-ddf420e0217e">

Note the error at the bottom of the terminal from when I did that (the generate tokens button):

<img width="1036" alt="Screenshot 2023-07-25 at 9 16 47 PM" src="https://github.com/netcreateorg/nc-multiplex/assets/5529898/3bbe4e65-d7b3-415e-a092-d2909e70b4ff">

**Comment by jdanish** [comment]
No worries, I don't need them because I can do on console... 

<img width="545" alt="Screenshot 2023-07-25 at 9 15 11 PM" src="https://github.com/netcreateorg/nc-multiplex/assets/5529898/861c261a-ba11-409d-a686-8ba0ac14e30d">

I ignore the errors because my understanding is that that comes from having run dev and not classroom, but classroom isn't working these days, so I live with it.

That token doesn't work though it should:

<img width="372" alt="Screenshot 2023-07-25 at 9 16 00 PM" src="https://github.com/netcreateorg/nc-multiplex/assets/5529898/3c057ca9-ee69-4f32-9ee9-8ebe8712bdb8">

I go to the other graph I started, and this time it works.

This other graph is the one I used with ./nc.js --dataset=....  when I built it from within netcreate-itest previously.

If I do this with more networks, only the one works. If I do each individually without multiplex, their tokens all work.

**Comment by jdanish** [comment]
One more observation - see the terminal out:

port 3001 is the one that is working, and 3002 is the one that is not. That seems to be the only one getting the pong not received error. 

```
...nc-multiplex: --> mapping to  testing 3002
SRV-NET  - socket connected
SRV-NET  - socket ADD UADDR_03 to network
SRV-NET  - SocketList change: add UADDR_03
SRV-NET  - 1 - UADDR_03
SRV-NET  - starting heartbeat
SRV-NET  - UADDR_03 FORWARD SRV_DBGET to 1 remotes
SRV-NET  - UADDR_03 RETURN SRV_DBGET from 1 remotes
...nc-multiplex: --> mapping to  testing 3002
SRV-NET  - adding 'SOURCE_UPDATE' reference to UADDR_03
SRV-NET  - adding 'NODE_DELETE' reference to UADDR_03
SRV-NET  - adding 'EDGE_UPDATE' reference to UADDR_03
SRV-NET  - adding 'EDGE_DELETE' reference to UADDR_03
SRV-NET  - adding 'EDIT_PERMISSIONS_UPDATE' reference to UADDR_03
SRV-NET  - adding 'NET_TEMPLATE_UPDATE' reference to UADDR_03
SRV-NET  - UADDR_03 FORWARD SRV_REG_HANDLERS to 1 remotes
SRV-NET  - UADDR_03 RETURN SRV_REG_HANDLERS from 1 remotes
SRV-NET  - UADDR_02 pong not received before time ran out -- CONNECTION DEAD!
...nc-multiplex: --> mapping to  ItestTeam 3001
SRV-NET  - socket DEL UADDR_03 from network
SRV-NET  - deleting 'SOURCE_UPDATE' reference to UADDR_03
SRV-NET  - deleting 'NODE_DELETE' reference to UADDR_03
SRV-NET  - deleting 'EDGE_UPDATE' reference to UADDR_03
SRV-NET  - deleting 'EDGE_DELETE' reference to UADDR_03
SRV-NET  - deleting 'EDIT_PERMISSIONS_UPDATE' reference to UADDR_03
SRV-NET  - deleting 'NET_TEMPLATE_UPDATE' reference to UADDR_03
SRV-NET  - SocketList change: del UADDR_03
...nc-multiplex: --> mapping to  ItestTeam 3001
SRV-NET  - socket connected
SRV-NET  - socket ADD UADDR_04 to network
SRV-NET  - SocketList change: add UADDR_04
SRV-NET  - 1 - UADDR_04
SRV-NET  - starting heartbeat
SRV-NET  - UADDR_04 FORWARD SRV_DBGET to 1 remotes
SRV-NET  - UADDR_04 RETURN SRV_DBGET from 1 remotes
...nc-multiplex: --> mapping to  ItestTeam 3001
SRV-NET  - adding 'SOURCE_UPDATE' reference to UADDR_04
SRV-NET  - adding 'NODE_DELETE' reference to UADDR_04
SRV-NET  - adding 'EDGE_UPDATE' reference to UADDR_04
SRV-NET  - adding 'EDGE_DELETE' reference to UADDR_04
SRV-NET  - adding 'EDIT_PERMISSIONS_UPDATE' reference to UADDR_04
SRV-NET  - adding 'NET_TEMPLATE_UPDATE' reference to UADDR_04
SRV-NET  - UADDR_04 [1-1-PEQ] FORWARD SRV_REG_HANDLERS to 1 remotes
SRV-NET  - UADDR_04 [1-1-PEQ] RETURN SRV_REG_HANDLERS from 1 remotes
SRV-NET  - UADDR_03 pong not received before time ran out -- CONNECTION DEAD!
```

**Comment by daveseah** [comment]
This issue is also closed by #2, so we can merge it into the main branch and clean up these lingering issues. Confirm?

**Comment by jdanish** [comment]
Confirmed!
